server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRedirect.java

```diff
@@ -1025,6 +1025,7 @@ protected void updateHeaders() throws MessagingException {
             setIsReply(newMail, isReply(originalMail), originalMail);
 
             newMail.getMessage().saveChanges();
+            newMail.removeAllAttributes();
```

Comment:
Ok but documentation doesn't run, you can't rely on it and it can very well be true at some time and becomes false because code is written by people that don't read or understand the documentation.
I agree that it may be the only solution, but could you check that there's no other option ?